crates/lib/src/protocols/types.rs

```diff
@@ -332,6 +334,25 @@ pub struct ExtraRequestSettings {
     pub check_app_id: Option<bool>,
 }
 
+/// Select how to go about gathering extra information via additional requests.
+///
+/// Used by:
+/// - [ExtraRequestSettings]
+/// - [valve::GatheringSettings]
+/// - [unreal2::GatheringSettings]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "clap", derive(clap::ValueEnum))]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Default)]
+pub enum GatherToggle {
+    /// Don't send a request.
+    #[default]
+    DontGather,
+    /// Send a request but ignore errors, instead return default or None.
+    AttemptGather,
+    /// Send a request and propagate any errors.
+    Required,
+}
```

Comment:
GatherToggle::DontGather
GatherToggle::AttemptGather
GatherToggle::Required

I think it would be cleaner to drop the 'Gather' suffix on the first 2 ones, and maybe afterwards rename 'Dont' to 'Skip', thoughts?

Reply:
Dont works. I'm not sure about Skip. Maybe Try or SkipOnFailure?

Reply:
what about this?

```rs
pub enum GatherToggle {
    /// No request will be sent. This option bypasses data gathering.
    #[default]
    Skip,

    /// A request will be sent, but errors are not critical. 
    /// In case of an error, the operation will return a default value or `None`.
    Try,

    /// A request will be sent, and any resulting errors will be propagated.
    /// This option treats successful data gathering as mandatory.
    Enforce,
}
```

Reply:
I like these.